src/read.c

```diff
@@ -1397,12 +1397,9 @@ static avifBool avifDecoderDataCopyTileToImage(avifDecoderData * data,
     const avifTile * firstTile = &data->tiles.tile[info->firstTileIndex];
     if (tile != firstTile) {
         // Check for tile consistency. All tiles in a grid image should match the first tile in the properties checked below.
-        avifBool uvPresent = (tile->image->yuvPlanes[AVIF_CHAN_U] && tile->image->yuvPlanes[AVIF_CHAN_V]);
-        avifBool firstTileUVPresent = (firstTile->image->yuvPlanes[AVIF_CHAN_U] && firstTile->image->yuvPlanes[AVIF_CHAN_V]);
         if ((tile->image->width != firstTile->image->width) || (tile->image->height != firstTile->image->height) ||
             (tile->image->depth != firstTile->image->depth) || (tile->image->yuvFormat != firstTile->image->yuvFormat) ||
-            (tile->image->yuvRange != firstTile->image->yuvRange) || (uvPresent != firstTileUVPresent) ||
-            (tile->image->colorPrimaries != firstTile->image->colorPrimaries) ||
+            (tile->image->yuvRange != firstTile->image->yuvRange) || (tile->image->colorPrimaries != firstTile->image->colorPrimaries) ||
```

Comment:
This change is necessary because `firstTile->image->yuvPlanes[AVIF_CHAN_U]` and `firstTile->image->yuvPlanes[AVIF_CHAN_V]` are already invalid when a shared codec decodes the second tile of a grid image. The original code only tests whether these two pointers are null or not and doesn't dereference them, but it's still bad to use invalid pointers.

With the stealer tracking, `firstTile->image->yuvPlanes[AVIF_CHAN_U]` and `firstTile->image->yuvPlanes[AVIF_CHAN_V]` become null pointers when a shared codec decodes the second tile of a grid image. So we can no longer test `(uvPresent != firstTileUVPresent)`. But I think the `(tile->image->yuvFormat != firstTile->image->yuvFormat)` test is equivalent.